### PR TITLE
Mscr 116

### PR DESCRIPTION
**Pull request by ilya-wd**

### CODE REVIEW

File: src/main/java/fi/vm/yti/datamodel/api/v2/endpoint/Schema.java [hunk 1]
```diff
@@ -41,126 +47,128 @@
...
-    		StoredFile file = files.get(0);
+		try {
+			byte[] fileInBytes = file.getBytes();
+			if (schemaDTO.getFormat() == SchemaFormat.JSONSCHEMA) {
+				ValidationRecord validationRecord = JSONValidationService.validateJSONSchema(fileInBytes);
+
+				boolean validationStatus = validationRecord.isValid();
+				List<String> validationMessages = validationRecord.validationOutput();
+
+				if (validationStatus) {
```
Comment (jkesanie) [inline src/main/java/fi/vm/yti/datamodel/api/v2/endpoint/Schema.java]: Rename validatationStatus variable. It is difficult to understand whether the if block should be executed or not just by reading the line "if(validatationStatus)".  Compared to for example "if(isValidJSON)"

File: src/main/resources/schema_v4 copy
Comment (jkesanie) [inline src/main/resources/schema_v4 copy:1]: Do we need this file?

File: src/test/java/fi/vm/yti/datamodel/api/v2/service/JSONValidationServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		
+		return inputSchemaInByte;
+	}
+	
+	@Test
+	void testValidJSONSchema() throws Exception, IOException {
+		String validInputSchemaPath = "test_jsonschema_b2share.json";
+		// *** validate format (eg email, date etc) with regex? ***
+		
+		assertEquals(JSONValidationService.validateJSON(byteStreamFromPath(validInputSchemaPath)), new ValidationRecord(true, Arrays.asList()));
```
Comment (jkesanie) [inline src/test/java/fi/vm/yti/datamodel/api/v2/service/JSONValidationServiceTest.java]: validataJSON() method does not exist. Have you compiled the code? Does the tests run successfully?

File: src/test/java/fi/vm/yti/datamodel/api/v2/service/SchemaServiceTest.java [hunk 3]
```diff
@@ -41,3 +41,4 @@ void testTransforJSONSchemaToInternal() throws Exception, IOException {
 		
 	}
 }
+
```
Comment (jkesanie) [inline src/test/java/fi/vm/yti/datamodel/api/v2/service/SchemaServiceTest.java:44]: Clean up

**Review by jkesanie — CHANGES_REQUESTED** [review]

File: src/main/resources/schema_v4 copy
Comment (ilya-wd) [reply src/main/resources/schema_v4 copy:1]: not anymore

**Review by jkesanie — APPROVED** [review]

**Pull request merged**